etcdserver/api/membership/cluster.go

```diff
@@ -759,16 +759,21 @@ func ValidateClusterAndAssignIDs(lg *zap.Logger, local *RaftCluster, existing *R
 	if len(ems) != len(lms) {
 		return fmt.Errorf("member count is unequal")
 	}
-	sort.Sort(MembersByPeerURLs(ems))
-	sort.Sort(MembersByPeerURLs(lms))
 
 	ctx, cancel := context.WithTimeout(context.TODO(), 30*time.Second)
 	defer cancel()
 	for i := range ems {
-		if ok, err := netutil.URLStringsEqual(ctx, lg, ems[i].PeerURLs, lms[i].PeerURLs); !ok {
+		var err error
+		ok := false
+		for j := range lms {
+			if ok, err = netutil.URLStringsEqual(ctx, lg, ems[i].PeerURLs, lms[j].PeerURLs); ok {
+				lms[j].ID = ems[i].ID
+				break
+			}
+		}
+		if !ok {
 			return fmt.Errorf("unmatched member while checking PeerURLs (%v)", err)
```

Comment:
nit: the `err` is error encountered in comparing `ems[i]` with last `lms`. This error statement made more sense when there was only level loop, so we know which `member` is being compared.
perhaps we can say `members` instead of `member` and print `ems[i]` and complete `lms` in the error message.

Reply:
can do.

Reply:
thanks @dbavatar